backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/humanaethica/enrollment/EnrollmentRepository.java

```diff
@@ -17,4 +17,8 @@ public interface EnrollmentRepository extends JpaRepository<Enrollment, Integer>
 
     @Query("SELECT e FROM Enrollment e WHERE e.volunteer.id = :userId")
     List<Enrollment> getEnrollmentsForVolunteerId(Integer userId);
+
+    @Query("SELECT CASE WHEN EXISTS (SELECT 1 FROM Enrollment e WHERE e.activity.id = :activityId AND e.volunteer.id = :volunteerId) THEN TRUE ELSE FALSE END")
```

Comment:
Do not have such a complex query. It is enough to have a method that returns an enrolment given the activity and the volunteer. The the code checks if the result is null or not.